src/cryptonote_core/loki_name_system.cpp

```diff
@@ -439,57 +439,75 @@ bool validate_lns_name(mapping_type type, std::string const &name, std::string *
   std::stringstream err_stream;
   LOKI_DEFER { if (reason) *reason = err_stream.str(); };
 
-  size_t max_name_len = 0;
-  if (type == mapping_type::session)      max_name_len = lns::SESSION_DISPLAY_NAME_MAX;
+  bool const is_lokinet = is_lokinet_type(type);
+  size_t max_name_len   = 0;
+
+  if (is_lokinet)                         max_name_len = lns::LOKINET_DOMAIN_NAME_MAX;
+  else if (type == mapping_type::session) max_name_len = lns::SESSION_DISPLAY_NAME_MAX;
   else if (type == mapping_type::wallet)  max_name_len = lns::WALLET_NAME_MAX;
-  else if (is_lokinet_type(type))         max_name_len = lns::LOKINET_DOMAIN_NAME_MAX;
   else
   {
     if (reason) err_stream << "LNS type=" << type << ", specifies unhandled mapping type in name validation";
     return false;
   }
 
   // NOTE: Validate name length
-  if (check_condition((name.empty() || name.size() > max_name_len), reason, "LNS type=", type, ", specifies mapping from name -> value where the name's length=", name.size(), " is 0 or exceeds the maximum length=", max_name_len, ", given name=", name))
+  if (check_condition((name.empty() || name.size() > max_name_len), reason, "LNS type=", type, ", specifies mapping from name->value where the name's length=", name.size(), " is 0 or exceeds the maximum length=", max_name_len, ", given name=", name))
     return false;
 
   // NOTE: Validate domain specific requirements
-  if (type == mapping_type::session)
+  char const *char_preceeding_suffix = nullptr;
+  if (is_lokinet)
   {
-  }
-  else if (is_lokinet_type(type))
-  {
-    // Domain has to start with a letter or digit, and can have letters, digits, or hyphens in between and must end with a .loki
+
+    // LOKINET
+    // Domain has to start with an alphanumeric, and can have (alphanumeric or hyphens) in between must end with a '.loki'
```

Comment:
Also the character preceding the ".loki" has to be alphanumeric.  (The code and regex comment do this, but it's missing from the comment description here).